app/models/amusement_park.rb

```diff
@@ -1,3 +1,11 @@
 class AmusementPark < ApplicationRecord
   has_many :rides
+  has_many :mechanics
+  has_many :amusement_park_mechanics
+  has_many :mechanics, through: :amusement_park_mechanics
+
+  def park_mechanics
+    self.select(:mechanics).distinct
+    # self.joins(rides: :mechanics).select(:mechanics).where("mechanics.rides.id = amusement_park.rides").distinct
```

Comment:
Continue working on this query. You're looking for all the rides of this amusement park, and we already have the association of `.rides`, so we don't necessarily need to joins rides, but can joins in the mechanics